### PR TITLE
Adds login component and forms and submission button and UI components

### PR DESCRIPTION
**Pull request by lbkchen**
* Login component for veterans is complete but not hooked up to the requester yet until the other PR is merged
* Added `Form.js` for any form submissions that we will be using in the mobile app, can handle validations and render errors to the user
* Added `Button.js` which submits form data and displays success/error states 
* Added app colors to `colors.js`
* Added `BackgroundOverlay.js` which renders the rotated green/white BG that claire designed
* Added `RaisedContainer.js` which renders a simple white container with drop shadow that maybe we can reuse

Login
img_2218b128a6c7-1

Login success (usually will not see this screen cause it advances)
img_2331

Login failure (renders errors, maybe we can show error strings later too)
img_2332

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -6,11 +6,12 @@ import ConnectScreen from './src/screens/ConnectScreen';
...
 import ResponseScreen from './src/screens/ResponseScreen';
 import ProfileScreen from './src/screens/ProfileScreen';
+import LoginScreen from './src/screens/LoginScreen';
 
 
 const Navigator = TabNavigator({
-  Home: {
-    screen: HomeScreen,
+  Login: {
+    screen: LoginScreen,
```
Comment (nikitr) [inline App.js:14]: should we still have the navigator when the user has not logged in yet?

File: src/screens/LoginScreen.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    } else {
+      console.error("Error occurred.");
+      onFailure && onFailure();
+    }
+  }
+
+  render() {
+    return (
+      <BackgroundOverlay>
+
```
Comment (nikitr) [inline src/screens/LoginScreen.js]: remove space

File: src/components/BackgroundOverlay.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+      <View style={styles.baseContainer}>
+        <View style={styles.rotatedContainer} />
+        {this.props.children}
+      </View>
+    );
+  }
+}
+
+
+const styles = StyleSheet.create({
```
Comment (nikitr) [inline src/components/BackgroundOverlay.js:29]: is it good convention to keep the styles in the same file?

File: src/screens/LoginScreen.js [hunk 4]
```diff
@@ -36,10 +42,14 @@ export default class LoginScreen extends React.Component {
     }
   }
 
+  redirectToSignupScreen(event, onSuccess, onFailure) {
```
Comment (nikitr) [inline src/screens/LoginScreen.js]: when are we calling this function? are we adding the logic in a separate PR?

File: src/components/RaisedContainer.js [hunk 5]
```diff
@@ -1,3 +1,10 @@
+/**
+ * Simple reusable white container with box shadow for containing elements.
+ *
+ * @prop style       - style override
+ * @prop children    - children contained in this component
+ */
```
Comment (nikitr) [inline src/components/RaisedContainer.js:6]: awesome! :smile:

**Review by nikitr — COMMENTED** [review]
All these components look great! Do you want to hook it up to the requester in a separate PR?

File: src/components/BackgroundOverlay.js (hunk 3 above)
Comment (lbkchen) [reply src/components/BackgroundOverlay.js:29]: I'm not sure actually haha. I think it's more convenient to edit when we only have common styles in a separate folder but you can make the call

File: App.js (hunk 1 above)
Comment (lbkchen) [reply App.js:14]: nope you're right, I was planning on figuring that out once i hook up the requester in another PR

Comment (lbkchen) [reply App.js:14]: this is just me being lazy

File: src/components/BackgroundOverlay.js (hunk 3 above)
Comment (nikitr) [reply src/components/BackgroundOverlay.js:29]: makes sense, i think we can keep it in the files they belong to unless it is used by multiple components!

**Comment by nikitr** [comment]
LGTM

**Pull request merged**